module/apmgopgv10/hook.go

```diff
@@ -0,0 +1,96 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build go1.14
+
+package apmgopgv10 // import "go.elastic.co/apm/module/apmgopgv10"
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/go-pg/pg/v10"
+
+	"go.elastic.co/apm"
+	"go.elastic.co/apm/module/apmsql"
+	"go.elastic.co/apm/stacktrace"
+)
+
+func init() {
+	stacktrace.RegisterLibraryPackage("github.com/go-pg/pg/v10")
+}
+
+const elasticApmSpanKey = "go-apm-agent:span"
+
+// Instrument modifies db such that operations are hooked and reported as spans
+// to Elastic APM if they occur within the context of a captured transaction.
+//
+// If Instrument cannot instrument db, then an error will be returned.
+func Instrument(db *pg.DB) error {
+	qh := &queryHook{}
+	switch qh := ((interface{})(qh)).(type) {
+	case pg.QueryHook:
+		db.AddQueryHook(qh)
+		return nil
+	}
+	return errors.New("cannot instrument pg.DB, does not implement required interface")
```

Comment:
```suggestion
	db.AddQueryHook(&queryHook{})
	return nil
```

IIRC, this interface approach existed because the AddQueryHook method didn't exist at the beginning of the v9 cycle. AddQueryHook exists in all versions of v10, so we should just call the method directly -- then we get static type checking back again.

Reply:
Makes sense, sorry for the screw-up. I was replacing the wrong path locally and still using the old version when I tested it.